cavern/src/main/java/org/opencadc/cavern/files/FileAction.java

```diff
@@ -70,51 +70,90 @@
 import ca.nrc.cadc.rest.InlineContentHandler;
 import ca.nrc.cadc.rest.RestAction;
 import ca.nrc.cadc.util.PropertiesReader;
+
+import ca.nrc.cadc.util.StringUtil;
+import ca.nrc.cadc.vos.ContainerNode;
+import ca.nrc.cadc.vos.DataNode;
 import ca.nrc.cadc.vos.Direction;
+import ca.nrc.cadc.vos.LinkingException;
+import ca.nrc.cadc.vos.Node;
+import ca.nrc.cadc.vos.NodeLockedException;
+import ca.nrc.cadc.vos.NodeNotFoundException;
+import ca.nrc.cadc.vos.NodeNotSupportedException;
 import ca.nrc.cadc.vos.VOSURI;
 
+import ca.nrc.cadc.vos.server.LocalServiceURI;
+import ca.nrc.cadc.vos.server.PathResolver;
+import ca.nrc.cadc.vos.server.auth.VOSpaceAuthorizer;
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.nio.file.attribute.GroupPrincipal;
 import java.nio.file.attribute.UserPrincipalLookupService;
 import java.security.AccessControlException;
 
 import org.apache.log4j.Logger;
-import org.opencadc.cavern.PosixIdentityManager;
+import org.opencadc.cavern.FileSystemNodePersistence;
 
 /**
  *
  * @author majorb
+ * @author jeevesh
  */
 public abstract class FileAction extends RestAction {
     private static final Logger log = Logger.getLogger(FileAction.class);
 
+    // Key values needed for FileAction
+    private Direction direction;
```

Comment:
Can the concept of direction be moved to GetAction and PutAction.  If you have `if` statements in here on direction (there are some) that logic should be moved down.  They already know their direction.

Reply:
as discussed - getDirection made an abstract function in FileAction, so GetAction & PutAction implement it and  it can be called in FileAction for use in authenticating tokens & user credentials.